### PR TITLE
Update DevilutionX to 1.5.3

### PR DESCRIPTION
**Pull request by Peppersawce**
It took me a while but I got the latest stable release (1.5.3) of DevilutionX and its core assets running flawlessly, aside for the lack of ZeroTier multiplayer support. This new version, compared to the one available on HaikuDepot, provides a lot of new enhancements, including a Settings menu with a huge amount of options.

### CODE REVIEW

**Comment by Peppersawce** [comment]
If you guys are having any issues testing this because of a lack of datafiles, the engine can also run on the shareware version's data, more info about it here: https://github.com/diasurgical/devilution/issues/466
(I wish I could've found a direct download for spawn.mpq as the way to get it currently is downloading and installing the demo, which is probably only possible in Windows)

File: games-rpg/devilutionx/devilutionx-1.5.3.recipe [hunk 1]
```diff
@@ -22,26 +23,35 @@ USER_SETTINGS_FILES="
...
+	app:devilutionx = $portVersion
 	"
 REQUIRES="
 	haiku$secondaryArchSuffix
+	lib:libbz2$secondaryArchSuffix
+	lib:libfmt$secondaryArchSuffix
 	lib:libSDL2_2.0$secondaryArchSuffix
 	lib:libSDL2_mixer_2.0$secondaryArchSuffix
 	lib:libSDL2_ttf_2.0$secondaryArchSuffix
+	lib:libSDL2_image_2.0$secondaryArchSuffix
```
Comment (korli) [inline games-rpg/devilutionx/devilutionx-1.5.3.recipe:35]: This should come before mixer.

File: games-rpg/devilutionx/devilutionx-1.5.3.recipe [hunk 2]
```diff
@@ -22,26 +23,35 @@ USER_SETTINGS_FILES="
...
 BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
-	devel:libsdl2_2.0$secondaryArchSuffix
+	devel:libfmt$secondaryArchSuffix
+	devel:libgettextlib$secondaryArchSuffix
+	devel:libgtest$secondaryArchSuffix
+	devel:libSDL2_2.0$secondaryArchSuffix
 	devel:libSDL2_mixer_2.0$secondaryArchSuffix
 	devel:libSDL2_ttf_2.0$secondaryArchSuffix
+	devel:libSDL2_image_2.0$secondaryArchSuffix
```
Comment (korli) [inline games-rpg/devilutionx/devilutionx-1.5.3.recipe:48]: Same here

File: games-rpg/devilutionx/devilutionx-1.5.3.recipe [hunk 3]
```diff
@@ -22,26 +23,35 @@ USER_SETTINGS_FILES="
...
 	devel:libsodium$secondaryArchSuffix
 	"
 BUILD_PREREQUIRES="
 	cmd:cmake
 	cmd:gcc$secondaryArchSuffix
+	cmd:git
```
Comment (korli) [inline games-rpg/devilutionx/devilutionx-1.5.3.recipe:54]: Shouldn't be needed.

File: games-rpg/devilutionx/patches/devilutionx-1.5.3.patchset [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+diff --git a/Source/utils/paths.cpp b/Source/utils/paths.cpp
+index e803436..dc65172 100644
+--- a/Source/utils/paths.cpp
++++ b/Source/utils/paths.cpp
+@@ -113,6 +113,9 @@ const std::string &ConfigPath()
+
+ const std::string &AssetsPath()
+ {
++#if defined(__HAIKU__)
++	assetsPath = "/boot/system/data/devilutionx/";
```
Comment (korli) [inline games-rpg/devilutionx/patches/devilutionx-1.5.3.patchset:16]: This should use find_directory (see https://github.com/haikuports/haikuports/blob/c3d10adaa97ccb238a739915689e51d76a96c83c/app-emulation/x16-emulator/patches/x16_emulator-46.patchset#L29 for example)

Comment (Peppersawce) [reply games-rpg/devilutionx/patches/devilutionx-1.5.3.patchset]: Done, I don't know how "good" the code is as I don't know C++ and just adapted what was done in the patch you linked me, but it works :)

Comment (korli) [reply games-rpg/devilutionx/patches/devilutionx-1.5.3.patchset]: Looks good!

**Pull request merged**